Framework/Core/include/Framework/ASoA.h

```diff
@@ -2627,10 +2676,34 @@ class Filtered<Filtered<T>> : public FilteredBase<typename T::table_t>
     return operator*=(other.getSelectedRows());
   }
 
-  using FilteredBase<typename T::table_t>::sliceByCached;
+  auto sliceByCached(framework::expressions::BindingNode const& node, int value)
+  {
+    uint64_t offset = 0;
+    std::shared_ptr<arrow::Table> result = nullptr;
+    auto status = ((table_t*)this)->getSliceFor(value, node.name.c_str(), result, offset);
+    if (status.ok()) {
+      auto start = offset;
+      auto end = start + result->num_rows();
+      auto start_iterator = std::lower_bound(this->getSelectedRows().begin(), this->getSelectedRows().end(), start);
```

Comment:
Isn't this equal_range?

Reply:
I don't think so. We have different values of row numbers, not all of them selected, and the `start_iterator` might point at some number higher than `offset`. Then the `stop_iterator` is compared with `end`, which is a totally different number.

Reply:
We have two lower bounds here, while equal_range is lower an upper. Unrelated, I think we should move this sort of selection vector manipulation in a separate function, since it is now used in multiple places.